_data/changelogs/component-checkbox.yml

```
@@ -2,6 +2,14 @@ title: Checkbox
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added indeterminate styles for checkboxes.
+    summaryAdditional: Checkboxes will now display as indeterminate when you set `input.indeterminate = true` via JavaScript or add the `data-indeterminate` attribute. This is currently only a style addition and does not affect checkbox functionality.
```

Optional: One more question - wondering if we should announce the `data-attribute` here. Is this something that is good practice to use, or should users stick to the native `:indeterminate` state?
```suggestion
    summaryAdditional: Checkboxes will now display as indeterminate when you set `input.indeterminate = true` via JavaScript. This is currently only a style addition and does not affect checkbox functionality.
```

---

That's a good point. Is there anything we should consider when setting `indeterminate` status via JS? You set the CSS pseudo-class via `input.indeterminate = true` 

Is there a JS benefit to allowing the use of a data attribute as well? 

If we deem that there isn't, should we remove support for the data attribute styles as well?

---

Initially, this seems fine to me. You can add the data attribute to get the initial styles first and then override when you need.

Something to keep in mind moving forward. Also totally possible that we revisit in the future once we have more information.